go/test/endtoend/vreplication/vdiff_helper_test.go

```diff
@@ -109,30 +124,46 @@ type expectedVDiff2Result struct {
 	hasMismatch bool
 }
 
-// todo: use specified cells
 func vdiff2(t *testing.T, keyspace, workflow, cells string, want *expectedVDiff2Result) {
 	ksWorkflow := fmt.Sprintf("%s.%s", keyspace, workflow)
 	t.Run(fmt.Sprintf("vdiff2 %s", ksWorkflow), func(t *testing.T) {
-		uuid, _ := performVDiff2Action(t, ksWorkflow, "create", "")
-		info := waitForVDiff2ToComplete(t, ksWorkflow, uuid)
+		uuid, _ := performVDiff2Action(t, ksWorkflow, cells, "create", "")
+		info := waitForVDiff2ToComplete(t, ksWorkflow, cells, uuid, time.Time{})
 
 		require.Equal(t, workflow, info.Workflow)
 		require.Equal(t, keyspace, info.Keyspace)
+		// I'm not sure if we always have rows in every table
+		//require.Greater(t, info.RowsCompared, int64(0))
 		if want != nil {
 			require.Equal(t, want.state, info.State)
 			require.Equal(t, strings.Join(want.shards, ","), info.Shards)
 			require.Equal(t, want.hasMismatch, info.HasMismatch)
 		} else {
-			require.Equal(t, info.State, "completed")
+			require.Equal(t, "completed", info.State)
 			require.False(t, info.HasMismatch)
-
 		}
+		if strings.Contains(t.Name(), "AcrossDBVersions") {
+			log.Errorf("VDiff resume cannot be guaranteed between major MySQL versions due to implied collation differences, skipping resume test...")
+			return
+		}
+		createCompletedTS, err := time.Parse(tsFormat, info.CompletedAt)
+		require.NoError(t, err)
+
```

Comment:
Will any of the "resumes" in the e2e tests see new data to diff or is the diff fully completed due to the previous "create" action?

Reply:
When using `--tablet_types=primary` and the `catchup()` function to ensure we get past the copy phase we don't process any additional rows. The vreplication_* workflows were passing once I added missing catchup calls and added that flag to `VDiff --v2`, but then other tests e.g. some vstream_* tests started failing. Do you want me track all of those down?